### PR TITLE
GConstruct PR for Single GPU Tests

### PR DESCRIPTION
**Pull request by prateekdesai04**
*Description of changes:*
Using Gconstruct pipeline for all Single GPU tests

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: training_scripts/gsgnn_np/ml_nc_utext_glem.yml [hunk 1]
```diff
@@ -53,6 +53,6 @@ gsf:
     use_node_embeddings: false
   node_classification:
     target_ntype: "movie"
-    label_field: "genre"
+    label_field: "label"
```
Comment (classicsong) [inline training_scripts/gsgnn_np/ml_nc_utext_glem.yml:56]: ml_nc_utext_glem.yml is not used by any single GPU test. You should not modify this file.

File: training_scripts/gsgnn_np/ml_nc_utext.yaml [hunk 2]
```diff
@@ -48,6 +48,6 @@ gsf:
     use_node_embeddings: false
   node_classification:
     target_ntype: "movie"
-    label_field: "genre"
+    label_field: "label"
```
Comment (classicsong) [inline training_scripts/gsgnn_np/ml_nc_utext.yaml:51]: ml_nc_utext.yaml is not used by any single GPU test. You should not modify this file.

File: tests/end2end-tests/data_gen/process_movielens.py [hunk 3]
```diff
@@ -54,10 +54,14 @@
...
+labels = []
 for i in range(5, 24):
-    attrs.append(np.array(movie[i]))
-attrs = np.stack(attrs, axis=1)
+    labels.append(np.array(movie[i]))
+labels = np.stack(labels, axis=1)
+label_list = []
+for i in range(labels.shape[0]):
+    label_list.append(np.nonzero(labels[i])[0][0])
+labels = np.array(label_list)
```
[the remark this answers is not in the document]
Comment (prateekdesai04) [reply tests/end2end-tests/data_gen/process_movielens.py:64]: We get the first non zero value and consider it as primary Genre as there are multiple genre labels from column 5 to 23. 
This change is borrowed from an old PR: https://github.com/awslabs/graphstorm/pull/141

File: tests/end2end-tests/create_data.sh [hunk 4]
```diff
@@ -1,178 +1,106 @@
...
 
+# movielens edge classification without data split - used in both single and multi-gpu tests
 rm -Rf /data/movielen_100k_multi_label_ec
 cp -R /data/movielen_100k_ec_1p_4t /data/movielen_100k_multi_label_ec
 python3 $GS_HOME/tests/end2end-tests/data_gen/gen_multilabel.py --path /data/movielen_100k_multi_label_ec --node_class false --field rate
 
-
 # Generate movielens dataset in gconstruct input format
-python3 $GS_HOME/tests/end2end-tests/data_process/process_movielens.py
+python3 $GS_HOME/tests/end2end-tests/data_gen/process_movielens.py
```
Comment (classicsong) [inline tests/end2end-tests/create_data.sh:91]: Why are you calling this command twice? L8 and L91?

File: tests/end2end-tests/data_gen/movielens_custom.json [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+                    "labels":	[
+                        {
+                            "label_col":	"label",
+                            "task_type":	"classification",
+                            "split_pct":	[0.8, 0.1, 0.1]
+                        }
+                    ]
+            }
+    ],
+    "edges": [
```
Comment (classicsong) [inline tests/end2end-tests/data_gen/movielens_custom.json:39]: Do we need labels on edges for movielens-custom graph?

File: tests/end2end-tests/data_gen/movielens_text.json [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+                    "labels":	[
+                        {
+                            "label_col":	"label",
+                            "task_type":	"classification",
+                            "split_pct":	[0.8, 0.1, 0.1]
+                        }
+                    ]
+            }
+    ],
+    "edges": [
```
Comment (classicsong) [inline tests/end2end-tests/data_gen/movielens_text.json:38]: Are you using the same graph for node classification and edge claasification?

File: tests/end2end-tests/data_gen/process_movielens.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (classicsong) [reply tests/end2end-tests/data_gen/process_movielens.py:64]: Can you put some comments to explain what are you doing here?

File: tests/end2end-tests/data_process/movielens_test.sh [hunk 7]
```diff
@@ -22,13 +22,13 @@ error_and_exit () {
 
 # Test the DistDGL graph format.
 echo "********* Test the DistDGL graph format ********"
-python3 $GS_HOME/tests/end2end-tests/data_process/process_movielens.py
-python3 -m graphstorm.gconstruct.construct_graph --conf-file $GS_HOME/tests/end2end-tests/data_process/movielens.json --num-processes 1 --output-dir /tmp/movielens --graph-name ml --add-reverse-edges --part-method random
+python3 $GS_HOME/tests/end2end-tests/data_gen/process_movielens.py
+python3 -m graphstorm.gconstruct.construct_graph --conf-file $GS_HOME/tests/end2end-tests/data_gen/movielens_text.json --num-processes 1 --output-dir /tmp/movielens --graph-name ml --add-reverse-edges --part-method random
```
Comment (classicsong) [inline tests/end2end-tests/data_process/movielens_test.sh:26]: Why the output-dir of movielens_text.json is /tmp/movielens but output-dir of data_gen/movielens.json is /tmp/movielens_bert?

Should output-dir of movielens_text.json be /tmp/movielens_with_tokens ?
And output-dir of movielens_text.json be /tmp/movielens_bert_emb?

File: tests/end2end-tests/graphstorm-nc/test.sh [hunk 8]
```diff
@@ -134,15 +134,15 @@ error_and_exit $?
...
-python3 $GS_HOME/tests/end2end-tests/data_gen/gen_multilabel.py --path /data/movielen_100k_multi_label_nc --node_class 1 --field genre
+python3 $GS_HOME/tests/end2end-tests/data_gen/gen_multilabel.py --path /data/movielen_100k_multi_label_nc --node_class 1 --field label
 
 echo "**************dataset: multilabel MovieLens, RGCN layer: 1, node feat: generated feature, inference: mini-batch, save emb"
-python3 -m graphstorm.run.gs_node_classification --workspace $GS_HOME/training_scripts/gsgnn_np --num-trainers $NUM_TRAINERS --num-servers 1 --num-samplers 0 --part-config /data/movielen_100k_multi_label_nc/movie-lens-100k.json --ip-config ip_list.txt --ssh-port 2222 --cf ml_nc.yaml --save-embed-path ./model/ml-emb/ --num-epochs 3 --multilabel true --num-classes 6 --node-feat-name feat
+python3 -m graphstorm.run.gs_node_classification --workspace $GS_HOME/training_scripts/gsgnn_np --num-trainers $NUM_TRAINERS --num-servers 1 --num-samplers 0 --part-config /data/movielen_100k_multi_label_nc/movie-lens-100k.json --ip-config ip_list.txt --ssh-port 2222 --cf ml_nc.yaml --save-embed-path ./model/ml-emb/ --num-epochs 3 --multilabel true --num-classes 6 --node-feat-name movie:title
```
Comment (classicsong) [inline tests/end2end-tests/graphstorm-nc/test.sh:140]: Do user nodes have feature? should it be movie:title user:feat?

File: tests/end2end-tests/data_process/movielens_test.sh (hunk 7 above)
Comment (prateekdesai04) [reply tests/end2end-tests/data_process/movielens_test.sh]: renamed it

File: tests/end2end-tests/data_gen/movielens_text.json (hunk 6 above)
Comment (prateekdesai04) [reply tests/end2end-tests/data_gen/movielens_text.json:38]: Yes

**Review by classicsong — APPROVED** [review]
LGTM

**Review by classicsong — APPROVED** [review]

**Pull request merged**